website/source/guides/secret-mgmt/versioned-kv.html.md

```diff
@@ -0,0 +1,950 @@
+created_time     2018-04-16T00:17:23.930473344Z
+deletion_time    n/a
+destroyed        false
+```
+
+In this example, the current version is 6. Notice that version 1 and 2 does not
+show up in the metadata. Because the kv secret engine is configured to keep only
+4 versions, the oldest two versions are permanently deleted that you won't be
+able to read it.
+
+```plaintext
+$ vault kv get -version=1 secret/customer/acme
+No value found at secret/data/customer/data
+```
+
+#### API call using cURL
+
+To set the `secret/` to keep up to 4 versions:
+
+```plaintext
+$ tee payload.json<<EOF
+{
+  "max_versions": 4,
+  "cas_required": false
+}
+EOF
+
+$ curl --header "X-Vault-Token: ..." \
+       --request POST \
+       --data @payload.json
+       http://127.0.0.1:8200/v1/secret/config
+```
+
+To view the configuration:
+
+```plaintext
+$ curl --header "X-Vault-Token: ..." \
+       http://127.0.0.1:8200/v1/secret/config | jq
+{
+ "request_id": "8addfed1-41eb-6a19-8342-93f493c51538",
+ "lease_id": "",
+ "renewable": false,
+ "lease_duration": 0,
+ "data": {
+   "cas_required": false,
+   "max_versions": 4
+ },
+ "wrap_info": null,
+ "warnings": null,
+ "auth": null
+}
+```
+
+Alternatively, to limit the number of versions only on the
+**`secret/customer/acme`** path rather than the entire `secret/` engine:
+
+```plaintext
+$ curl --header "X-Vault-Token: ..." \
+       --request POST \
+       --data @payload.json
+       http://127.0.0.1:8200/v1/secret/metadata/customer/acme
+```
+
+Invoke the `secret/metadata/customer/acme` endpoint instead.
+
+
+Overwrite the data a few more times to see what happens to the data.
+
+```plaintext
+$ curl --header "X-Vault-Token: ..." \
+       http://127.0.0.1:8200/v1/secret/metadata/customer/acme | jq
+{
+ "request_id": "f2dd7f69-294c-e5c3-d582-f723005ea243",
+ "lease_id": "",
+ "renewable": false,
+ "lease_duration": 0,
+ "data": {
+   "created_time": "2018-04-14T00:42:25.677078177Z",
+   "current_version": 6,
+   "max_versions": 0,
+   "oldest_version": 3,
+   "updated_time": "2018-04-16T00:17:23.930473344Z",
+   "versions": {
+     "3": {
+       "created_time": "2018-04-16T00:15:59.880368849Z",
+       "deletion_time": "",
+       "destroyed": false
+     },
+     "4": {
+       "created_time": "2018-04-16T00:16:18.941331243Z",
+       "deletion_time": "",
+       "destroyed": false
+     },
+     "5": {
+       "created_time": "2018-04-16T00:16:34.407951572Z",
+       "deletion_time": "",
+       "destroyed": false
+     },
+     "6": {
+       "created_time": "2018-04-16T00:17:23.930473344Z",
+       "deletion_time": "",
+       "destroyed": false
+     }
+   }
+ },
+ "wrap_info": null,
+ "warnings": null,
+ "auth": null
+}
+```
+
+In this example, the current version is 6. Notice that version 1 and 2 does not
+show up in the metadata. Because the kv secret engine is configured to keep only
+4 versions, the oldest two versions are permanently deleted that you won't be
+able to read it.
```

Comment:
the oldest two versions are permanently deleted and you won't be able to read them